### PR TITLE
[PyOV] Try to fix PosixPath error in CI

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - Added `tmp_path` [fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture) for working with temporary files in tests.
 - Update `test_manager.py` -- reduce amount of `DeprecationWarning` (use Serialize transformation directly wherever it's possible).
 - Update `Serialize` python ctor -- support of `IR Vesrion` enum in python + add `string`-type for `IR Version` to be aligned with `manager.register_path` for Serialize transformation.
 - Move `convert_to_version` under `Common::utils` namespace.

### Tickets:
 - 98936

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/graph/passes/transformations.cpp [hunk 1]
```diff
@@ -20,6 +20,19 @@
 #include "pyopenvino/utils/utils.hpp"
 
 namespace py = pybind11;
+using Version = ov::pass::Serialize::Version;
+
+
+inline Version convert_to_version(const std::string& version) {
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/passes/transformations.cpp:26]: like placeholder.
Have the same function on `manager.cpp` -- move to `utils.cpp`.

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/graph/passes/transformations.cpp]: done

**Review by jiwaszki — APPROVED** [review]
LGTM, @p-wysocki can you take a look?

**Review by p-wysocki — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
is it applicable to port this to 2022.3?

**Pull request merged**